packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -835,14 +843,27 @@ function commitWork(current: Fiber | null, finishedWork: Fiber): void {
     case Profiler: {
       if (enableProfilerTimer) {
         const onRender = finishedWork.memoizedProps.onRender;
-        onRender(
-          finishedWork.memoizedProps.id,
-          current === null ? 'mount' : 'update',
-          finishedWork.actualDuration,
-          finishedWork.treeBaseDuration,
-          finishedWork.actualStartTime,
-          getCommitTime(),
-        );
+
+        if (enableInteractionTracking) {
+          onRender(
+            finishedWork.memoizedProps.id,
+            current === null ? 'mount' : 'update',
+            finishedWork.actualDuration,
+            finishedWork.treeBaseDuration,
+            finishedWork.actualStartTime,
+            getCommitTime(),
+            Array.from((root.memoizedInteractions: any)),
```

Comment:
Why pay the cost of cloning this data? Why not just pass the set and let the consumer decide?

Reply:
`memoizedInteractions` is a `Set` and the I initially proposed passing `onRender` an `Array`. We could change this so that the API passes a `Set` instead, then I wouldn't _need_ to clone it.

Users could mutate it though, and I think a `Set` can't be object-frozen...?

Open to suggestions here 😄